frontend/src/views/assignments/offer-actions.js

```diff
@@ -147,6 +230,38 @@ function OfferActionButtons(props) {
             >
                 Set as Rejected
             </ActionButton>
+            <Modal
```

Comment:
Make the entire modal a component. Then you won't need to separate out the Table as well. This is also getting big enough it should probably be in its own file.